source/agora/consensus/data/Block.d

```diff
@@ -551,18 +554,55 @@ public Block makeNewBlock (Transactions)(const ref Block prev_block,
     return block;
 }
 
+/***************************************************************************
+
+    Create the block random seed from the provided pre-images
+
+    It might be that not all validators provide their `pre-image` and in
+    that case the length of the preimages can be less than the number of
+    active validators.
+
+    Params:
+        preimages = preimages sorted by the utxo keys of the validators
+
+    Returns:
+        Hash of all provided pre-images or Hash.init if an exception is thrown
+
+***************************************************************************/
+
+
+public static Hash createRandomSeed (Preimages)(Preimages preimages) @safe nothrow
+{
+    try
+    {
+        if (preimages.count == 0)
+            assert(0, "createRandomSeed: No preimages to create block random seed!");
+        return preimages.reduce!((a , b) => hashMulti(a, b));
+    } catch (Exception e)
+    {
+        log.error("createRandomSeed: Exception thrown {}", e);
+        return Hash.init;
+    }
+}
```

Comment:
Why not let it throw ?

Reply:
I have added a commit to handle the exception at higher levels.